account_clearance_plan/wizard/account_clearance_plan.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountClearancePlanLine(models.TransientModel):
+    _name = "account.clearance.plan.line"
+    _description = "Clearance Plan Line"
+
+    clearance_plan_id = fields.Many2one(
+        comodel_name="account.clearance.plan", required=True
+    )
+    amount = fields.Float(required=True)
+    date_maturity = fields.Date(string="Due Date", required=True)
+
+    @api.constrains("amount")
+    def _check_positive_amount(self):
+        for rec in self:
+            if rec.amount < 0:
+                raise Warning(_("Amounts should all be positive."))
+
+
+class AccountClearancePlan(models.TransientModel):
+    _name = "account.clearance.plan"
+    _description = "Clearance Plan"
+
+    move_line_ids = fields.Many2many("account.move.line", readonly=True)
+    journal_id = fields.Many2one(
+        string="Journal",
+        comodel_name="account.journal",
+        required=True,
+        help="Journal of the new entry.",
+    )
+    amount_to_allocate = fields.Float(string="Total Amount to Allocate", readonly=True)
+    amount_allocated = fields.Float(
+        string="Amount Allocated", compute="_compute_amount_allocated"
+    )
+    clearance_plan_line_ids = fields.One2many(
+        comodel_name="account.clearance.plan.line", inverse_name="clearance_plan_id"
+    )
```

Comment:
Do we need to populate `ref`, `narration` in the generated move?

Reply:
It's quite difficult to give a ref that makes sense, the clearance plan may be created based on many invoices or move lines. Do you have any suggestion ?

I implemented the rest of your comments.

Reply:
Can the user provide it?

Reply:
Indeed !
The user can now provide it in the wizard as well as the `narration`.